Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/listitem/businessrule/BusinessRuleVH.kt

```diff
@@ -42,10 +46,22 @@ class BusinessRuleVH(
         }
     }
 
+    private fun countryName(countryCode: String?, userLocale: Locale = Locale.getDefault()): String =
+        if (countryCode != null) {
+            Locale(userLocale.language, countryCode.uppercase()).getDisplayCountry(userLocale)
+        } else ""
+
+    fun DccValidationRule.getRuleDescription(): String {
+        val currentLocaleCode = Locale.getDefault().language
+        val descItem = description.find { it.languageCode == currentLocaleCode }
+            ?: description.find { it.languageCode == "en" } ?: description.firstOrNull()
+        return descItem?.description ?: identifier
+    }
+
     data class Item(
         @DrawableRes val ruleIconRes: Int,
-        val ruleDescriptionText: LazyString,
-        val countryInformationText: LazyString,
+        val ruleDescriptionText: DccValidationRule,
+        val countryInformationText: Pair<Int, String?>,
```

Comment:
since you are already passing the `rule` do you still need to  this `pair` ?  I suggest you do the following check in `VH` directly.

```kotlin
when(item.rule.type){
  DccValidationRule.Type.ACCEPTANCE ->
            context.getString(R.string.validation_rules_acceptance_country,
            country)
   DccValidationRule.Type.INVALIDATION -> 
          context.getString( R.string.validation_rules_invalidation_country)
}           
```

what do you think?

Reply:
Yes, I was thinking about that too. I'll push an update with the code moved.